### ISSUE TITLE
Inlining self members is not working

### ISSUE THREAD

**Issue by kobbled** [kobbled/tp_plus#47]
Calling members from the same namespace and trying to inline them does not work:

**TP+**
```ruby
local := R[50..80]

namespace Calc

  inline def normalizeVector(n1,n2,n3,e1,e2,e3)
    norm := LR[]
    
    norm = Mth::SQRT(n1*n1 + n2*n2 + n3*n3)
    indirect('r', e1) = n1/norm
    indirect('r', e2) = n2/norm
    indirect('r', e3) = n3/norm
  end

  inline def dot(nx,ny,nz,vx,vy,vz,d)
      indirect('r', d) = nx*vx + ny*vy + nz*vz
  end

  inline def intersect(x1,y1,z1,x2,y2,z2,nx,ny,nz)
    ux := LR[]
    uy := LR[]
    uz := LR[]
    vx := LR[]
    vy := LR[]
    vz := LR[]
    d  := LR[]

    # vector from two points
    vx=x2-x1
    vy=y2-y1
    vz=z2-z1

    Calc::normalizeVector(vx,vy,vz,&ux,&uy,&uz)  # unit vector
    d = Calc::dot(nx,ny,nz,ux,uy,uz)  # dot product
  end
end

X1 := -0.30694321232747335
Y1 := 1.6723049456001353
Z1 := 1.432745154010752
X2 := -0.7868785787889008
Y2 := 1.7975569490378929
Z2 := 1.44473503116485
CX := -0.23966969549655914
CY := 1.6529096364974976
CZ := 1.438263177871704

Calc::intersect(X1,Y1,Z1,X2,Y2,Z2,CX,CY,CZ)

```

**Comment by kobbled** [comment]
Also dot() function is not compiling, giving this stack error:
```
localstack/stack.rb:54:in `add': undefined method `key?' for nil:NilClass (NoMethodError)

        unless @stack[0].key?(name)\r
                        ^^^^^
        from C:/Users/Matt Dewar/Dropbox/Fanuc_programs/tp_plus/lib/tp_plus/nodes/local_node.rb:22:in `getDefinitions'
        from C:/Users/Matt Dewar/Dropbox/Fanuc_programs/tp_plus/lib/tp_plus/nodes/reg_range_node.rb:14:in `eval'
        from C:/Users/Matt Dewar/Dropbox/Fanuc_programs/tp_plus/lib/tp_plus/interpreter.rb:304:in `preprocess_local_variables'
        from C:/Users/Matt Dewar/Dropbox/Fanuc_programs/tp_plus/lib/tp_plus/base_block.rb:309:in `call'
        from C:/Users/Matt Dewar/Dropbox/Fanuc_programs/tp_plus/lib/tp_plus/base_block.rb:309:in `block in traverse_nodes'
        from C:/Users/Matt Dewar/Dropbox/Fanuc_programs/tp_plus/lib/tp_plus/base_block.rb:241:in `each'
        from C:/Users/Matt Dewar/Dropbox/Fanuc_programs/tp_plus/lib/tp_plus/base_block.rb:241:in `each_with_index'
        from C:/Users/Matt Dewar/Dropbox/Fanuc_programs/tp_plus/lib/tp_plus/base_block.rb:241:in `traverse_nodes'
        from C:/Users/Matt Dewar/Dropbox/Fanuc_programs/tp_plus/lib/tp_plus/interpreter.rb:375:in `eval'
        from C:/Users/Matt Dewar/Dropbox/Fanuc_programs/tp_plus/lib/tp_plus/function.rb:157:in `interpret'
        from C:/Users/Matt Dewar/Dropbox/Fanuc_programs/tp_plus/lib/tp_plus/interpreter.rb:389:in `block in eval'
        from C:/Users/Matt Dewar/Dropbox/Fanuc_programs/tp_plus/lib/tp_plus/interpreter.rb:388:in `each'
        from C:/Users/Matt Dewar/Dropbox/Fanuc_programs/tp_plus/lib/tp_plus/interpreter.rb:388:in `eval'
        from C:/Users/Matt Dewar/Dropbox/Fanuc_programs/tp_plus/bin/tpp:167:in `<main>'
```

This error does not occur separating the function out from the namespace

**Comment by kobbled** [comment]
within the namespace refer to self functions without the namespace prefix like:

```
 normalizeVector(vx,vy,vz,&ux,&uy,&uz)  # unit vector
```
not
```
 Calc::normalizeVector(vx,vy,vz,&ux,&uy,&uz)  # unit vector
```

**Comment by kobbled** [comment]
should be fixed in commit 6b6d9f27be964da8796fc38386080b8b298bd21b. However, only two levels deep were tested.
